src/number_format.js

```diff
@@ -214,10 +219,13 @@ class NumberFormat extends React.Component {
   onKeyDown(e) {
     const el = e.target;
     const {selectionStart, selectionEnd, value} = el;
-    const {decimalPrecision} = this.props;
+    const {decimalPrecision, allowNegative} = this.props;
     const {key} = e;
     const numRegex = this.getNumberRegex(false, decimalPrecision !== false);
     //Handle backspace and delete against non numerical/decimal characters
+    if (key === '-' && allowNegative) {
+      this.setState({negative: !this.state.negative});
```

Comment:
Yes I will try to find time to do this - in my workplace I have a version that does this but it it integrated into an old version of this library.